birdstagram/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.create(user_params)
+    session[:user_id] = @user.id
+    if @user.save
+      redirect_to '/posts'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  private
+
+    def user_params
+      params.require(:user).permit(:username, :email, :password)
```

Comment:
Is this using strong params? This is my understanding where you are limiting or permitting input params.